pyirf/interpolation/component_estimators.py

```diff
@@ -0,0 +1,673 @@
+        grid_points: np.ndarray, shape=(n_points, n_dims):
+            Grid points at which interpolation templates exist
+        params: np.ndarray, shape=(n_points, ..., n_params)
+            Corresponding parameter values at each point in grid_points.
+            First dimesion has to correspond to number of grid_points.
+        interpolator_cls:
+            pyirf interpolator class, defaults to GridDataInterpolator.
+        interpolator_kwargs: dict
+            Dict of all kwargs that are passed to the interpolator, defaults to
+            None which is the same as passing an empty dict.
+        extrapolator_cls:
+            pyirf extrapolator class. Can be and defaults to ``None``,
+            which raises an error if a target_point is outside the grid
+            and extrapolation would be needed.
+        extrapolator_kwargs: dict
+            Dict of all kwargs that are passed to the extrapolator, defaults to
+            None which is the same as passing an empty dict.
+
+        Raises
+        ------
+        TypeError:
+            When interpolator_cls is not a ParametrizedInterpolator subclass.
+        TypeError:
+            When params is not a np.ndarray
+        ValueError:
+            When number of points grid_points and params is not matching
+
+        Note
+        ----
+            Also calls pyirf.component_estimators.BaseComponentEstimator.__init__
+        """
+
+        super().__init__(
+            grid_points,
+        )
+
+        if not isinstance(params, np.ndarray):
+            raise TypeError("Input params is not a numpy array.")
+        elif self.n_points != params.shape[0]:
+            raise ValueError(
+                "Shape missmatch, number of grid_points and rows in params not matching."
+            )
+
+        if interpolator_kwargs is None:
+            interpolator_kwargs = {}
+
+        if extrapolator_kwargs is None:
+            extrapolator_kwargs = {}
+
+        if not issubclass(interpolator_cls, ParametrizedInterpolator):
+            raise TypeError(
+                f"interpolator_cls must be a ParametrizedInterpolator subclass, got {interpolator_cls}"
+            )
+
+        self.interpolator = interpolator_cls(grid_points, params, **interpolator_kwargs)
+
+        if extrapolator_cls is None:
+            self.extrapolator = None
+        else:
+            self.extrapolator = extrapolator_cls(
+                grid_points, params, **extrapolator_kwargs
+            )
+
+
+class AEFFEstimator(ParametrizedComponentEstimator):
+    @u.quantity_input(effective_area=u.m**2, min_effective_area=u.m**2)
+    def __init__(
+        self,
+        grid_points,
+        effective_area,
+        interpolator_cls=GridDataInterpolator,
+        interpolator_kwargs=None,
+        extrapolator_cls=None,
+        extrapolator_kwargs=None,
+        min_effective_area=1 * u.m**2,
+    ):
+        """
+        Estimator class for effective areas. Takes a grid of effective areas
```

Comment:
Add a newline and empty line after `Estimator class for effective area`.

The first line should be a short "title" for the docs to look nice. Also for the others.